Applications/ClientWebPortal/Source/webapp/src/utils/chainable-models/indexed-model.js

```diff
@@ -0,0 +1,76 @@
+import ArrayModel from './array-model';
+import HashMap from 'hashmap';
+import ArrayStringKeyGenerator from 'utils/array-string-key-generator';
+
+export default class {
+  constructor(indices, columnNames, keyGenerator = new ArrayStringKeyGenerator()) {
+    // validate indices
+    let columnLength = columnNames.length;
+    if (indices == null || indices.length > columnNames) {
```

Comment:
This check isn't needed.  Your else branch will cover this scenario.

Reply:
Once it enters the else without the null check, it would throw error at "indices.length".

Reply:
Yes, that is correct behavior.

Reply:
Ah ok, I see what you mean.